src/types.rs

```diff
@@ -0,0 +1,49 @@
+//! Concrete node and edge types used in the registry.
+
+/// The type of a node.
+pub enum NodeType {
+    /// A user, eg. contributor, project member etc.
+    User,
+    /// A project with users as members and contributors.
+    Project,
+}
+
+/// Node data.
+type NodeData = NodeType;
+
+/// The type of an edge.
+pub enum EdgeType {
+    /// Contribution, user -> project.
+    ContributionToProject,
+    /// Contribution, project -> user.
+    ContributionFromUser,
+    /// Membership of a user in a project, project <-> user.
+    ProjectMembership,
+    /// One-way dependency between two projects, project -> project.
+    ProjectDependency,
+}
+
+/// Edge data.
+pub struct EdgeData {
+    /// The type of edge.
+    edge_type: EdgeType,
+    /// A strength factor for this edge. Eg. the number of contributions
+    /// on a contribution edge.
+    strength: f64,
+}
```

Comment:
With this declaration `strenght` is an attribute of all edges. It is unclear to me what the meaning of `strength` is for Dependencies. Why not have `strength` only for Contributions and rename it to `count` to make its meaning explicit?

Reply:
I had it like that previously, but since it's the same strength in both directions, and the data is shared, it made more sense to extract it. It can also be used for dependencies if we want to. Renaming it to `count` limits how we think about it.

Reply:
Another shower thought: if `strength` was a polymorphic parameter, then you could use `Option<..>` in some circumstances where you are not interested in this attribute at all, i.e.:

``` rust
pub struct EdgeData<S> {
    edge_type: EdgeType,
    strength: S
}
```

And later for a specific type have `EdgeData<S = Option<f64>>`, but I do appreciate that this comes with the extra burden of the type parameter, which might not be desirable. 

> Renaming it to count limits how we think about it.

Why not simply call the field `data`? I agree with @geigerzaehler here -- I find `strength` not very communicative. If we want to use something really generic, perhaps `data` is as generic as we can think 😉

Reply:
I've renamed it to `weight` and improved the comment.

Reply:
I am fine with `weight` (I know this is pretty much 🚲 - shedding, but names are still important to be able to think about things clearly) -- if anything, the fact that an edge has a `weight` is a concept which should be familiar to most engineers, more than the one of `strength` :)

Reply:
Yeah exactly. I don't mind the discussions around names - there's nothing worse than picking a confusing name which results in the same questions/bugs over and over because of wrong assumptions.

Reply:
If we use a generic `data` attribute, don’t we loose the concrete information about the contribution count? For the ledger it is important to know about the concrete types. We don’t care so much about the abstract underpinning.

Reply:
As I understood our discussion Osrank will take care of determining the weights from data the ledger gives it. This would mean that the Ledger would provide the count and osrank picks it up and turns it into a weight.

(My post above was written before I saw your answer, @cloudhead, since my Github UI did not update. This is why it might seem out of place)